pkg/controllers/deprovisioning/controller.go

```diff
@@ -110,17 +110,20 @@ func (c *Controller) Reconcile(ctx context.Context, _ reconcile.Request) (reconc
 	// with making any scheduling decision off of our state nodes. Otherwise, we have the potential to make
 	// a scheduling decision based on a smaller subset of nodes in our cluster state than actually exist.
 	if !c.cluster.Synced(ctx) {
+		logging.FromContext(ctx).Debugf("waiting for in-memory cluster state to sync before deprovisioning")
 		return reconcile.Result{RequeueAfter: time.Second}, nil
 	}
 	// Attempt different deprovisioning methods. We'll only let one method perform an action
 	isConsolidated := c.cluster.Consolidated()
 	for _, d := range c.deprovisioners {
+		ctx = context.WithValue(ctx, "deprovisioner", d.String())
```

Comment:
This adds it to the context, not the logger.

Reply:
Let's inject it into the logger instead of the context.